### PR TITLE
restore db instance fom db snapshot

### PR DESCRIPTION
**Pull request by michelleN**
- add mock
- add restore method on servers collection
- modify save method on server model

### CODE REVIEW

**Comment by lanej** [comment]
real tests errored:

```
    #restore - succeeds
    InvalidDBInstanceState => Cannot create a snapshot because the database instance fog-snapshot-test-6da4 is not currently in the available state. (Fog::AWS::RDS::Error)
      /Users/jlane/.gem/ruby/2.2.2/gems/excon-0.45.3/lib/excon/middlewares/expects.rb:6:in `response_call'
      /Users/jlane/.gem/ruby/2.2.2/gems/excon-0.45.3/lib/excon/middlewares/response_parser.rb:8:in `response_call'
      /Users/jlane/.gem/ruby/2.2.2/gems/excon-0.45.3/lib/excon/connection.rb:372:in `response'
      /Users/jlane/.gem/ruby/2.2.2/gems/excon-0.45.3/lib/excon/connection.rb:236:in `request'
      /Users/jlane/.gem/ruby/2.2.2/gems/fog-xml-0.1.2/lib/fog/xml/sax_parser_connection.rb:35:in `request'
      /Users/jlane/.gem/ruby/2.2.2/gems/fog-xml-0.1.2/lib/fog/xml/connection.rb:7:in `request'
      /Users/jlane/p/fog-aws/lib/fog/aws/rds.rb:254:in `_request'
      /Users/jlane/p/fog-aws/lib/fog/aws/rds.rb:249:in `request'
```

**Comment by geemus** [comment]
@michelleN Looks like this is failing in CI now, could you take another look? Thanks!

File: lib/fog/aws/requests/rds/restore_db_instance_from_db_snapshot.rb [hunk 1]
```diff
@@ -20,8 +20,74 @@ def restore_db_instance_from_db_snapshot(snapshot_id, db_name, opts={})
...
+            "DBSecurityGroups"                 => [{'Status'=>'active', 'DBSecurityGroupName'=>'default'}],
+            "Endpoint"                         => {},
+            "Engine"                           => options['Engine'] || snapshot['Engine'],
+            "EngineVersion"                    => options['EngineVersion'] || snapshot['EngineVersion'],
+            "InstanceCreateTime"               => nil,
+            "Iops"                             => options['Iops'],
+            "LicenseModel"                     => options['LicenseModel'] || snapshot['LicenseModel'] || 'general-public-license',
+            "MasterUsername"                   => options['MasterUsername'] || snapshot['MasterUsername'],
+            "MultiAZ"                          => !!multi_az,
+            "OptiongroupMemberships"           => option_group_membership,
```
Comment (lanej) [inline lib/fog/aws/requests/rds/restore_db_instance_from_db_snapshot.rb]: Seems like a typo. Should be `OptionGroupMemberships`

File: lib/fog/aws/requests/rds/restore_db_instance_to_point_in_time.rb [hunk 2]
```diff
@@ -21,7 +21,51 @@ def restore_db_instance_to_point_in_time(source_db_name, target_db_name, opts={}
...
+            "DBSecurityGroups"                 => [{'Status'=>'active', 'DBSecurityGroupName'=>'default'}],
+            "Endpoint"                         => {},
+            "Engine"                           => opts['Engine'] || 'mysql',
+            "EngineVersion"                    => opts['EngineVersion'] || '5.5.6',
+            "InstanceCreateTime"               => nil,
+            "Iops"                             => opts['Iops'],
+            "LicenseModel"                     => 'general-public-license',
+            "MasterUsername"                   => 'username',
+            "MultiAZ"                          => !!opts['MultiAZ'],
+            "OptiongroupMemberships"           => [{ 'OptionGroupMembership' => [{ 'OptionGroupName' => 'default: mysql-5.6', 'Status' => "pending-apply"}] }],
```
Comment (lanej) [inline lib/fog/aws/requests/rds/restore_db_instance_to_point_in_time.rb]: same mistype here maybe?

**Comment by lanej** [comment]
Stale.  Closing.

**Pull request closed without merging**